pkgs/tools/backup/autorestic/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib, fetchFromGitHub, installShellFiles, buildGoModule }:
+
+buildGoModule rec {
+  name = "autorestic";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "cupcakearmy";
+    repo = "${name}";
+    rev = "v${version}";
+    sha256 = "yQgSJ0SQNWPMyrYn8rep+1b549HP8sOERh+kOiAK3+c=";
+  };
+
+  vendorSha256 = "7648gAguqeqLKFS9xRcx20wpSLb+ykZ7rOqR5PKe71o=";
+
+  subPackages = [ "." ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
+    installShellCompletion --cmd autorestic \
+         --bash <($out/bin/autorestic completion bash) \
+         --fish <($out/bin/autorestic completion fish) \
+         --zsh <($out/bin/autorestic completion zsh) \
+  '';
+
+  meta = with lib; {
+    description = "High level CLI utility for restic";
+    homepage = "https://github.com/cupcakearmy/autorestic";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ renesat ];
+    platforms = platforms.linux ++ platforms.darwin;
```

Comment:
Set by buildGoModule. Only set this unless it really doesn't support anything else. But i wouldn't know why it couldn't support FreeBSD.

Reply:
I do not add FreeBSD because in package repo install script using only Linux and Darwin and in restic package usings this platforms.